### PR TITLE
GearList 

### PR DESCRIPTION
**Pull request by EllieLilithFox**
Convert GearList component to functional component and add tests, add types for gear-list actions

### CODE REVIEW

File: gearcatalog/frontend/src/actions/actions.types.ts [hunk 1]
```diff
@@ -1,2 +1,21 @@
+import { Gear } from '../components/gear/types';
+
+type GearNoId = {
```
Comment (NatalieWolfe) [inline gearcatalog/frontend/src/actions/actions.types.ts]: What is the use case for this type?

File: gearcatalog/frontend/src/components/gear/EditGearForm.tsx [hunk 2]
```diff
@@ -34,7 +35,7 @@ match: { params } }: EditGearProps) => {
     useInput(selectedGear.locking);
 
   useEffect(() => {
-    selectGear(params.id, token);
+    selectGear(params.id.toString(), token);
```
Comment (NatalieWolfe) [inline gearcatalog/frontend/src/components/gear/EditGearForm.tsx]: Why string-ify the id?

**Review by NatalieWolfe — CHANGES_REQUESTED** [review]

File: gearcatalog/frontend/src/actions/actions.types.ts (hunk 1 above)
Comment (EllieLilithFox) [reply gearcatalog/frontend/src/actions/actions.types.ts]: Adding new gear.

File: gearcatalog/frontend/src/components/gear/EditGearForm.tsx (hunk 2 above)
Comment (EllieLilithFox) [reply gearcatalog/frontend/src/components/gear/EditGearForm.tsx]: selectGear takes a string in the first argument. That's what it is typed as.

File: gearcatalog/frontend/src/actions/actions.types.ts (hunk 1 above)
Comment (NatalieWolfe) [reply gearcatalog/frontend/src/actions/actions.types.ts]: Could you document its use in the code? As it is it seems a bit odd.

File: gearcatalog/frontend/src/components/gear/EditGearForm.tsx (hunk 2 above)
Comment (NatalieWolfe) [reply gearcatalog/frontend/src/components/gear/EditGearForm.tsx]: Why? If the first argument is the ID and IDs are numbers, shouldn't it be a number?

Comment (EllieLilithFox) [reply gearcatalog/frontend/src/components/gear/EditGearForm.tsx]: Yeah, that makes sense. I'll change the type on the functions.

File: gearcatalog/frontend/src/actions/actions.types.ts (hunk 1 above)
Comment (EllieLilithFox) [reply gearcatalog/frontend/src/actions/actions.types.ts]: I need a gear with no id when adding new gear to the database. I went ahead and moved it to the gear types file and added a comment for it explaining why it's there.

**Pull request merged**